### PR TITLE
Docker compose testing

### PR DESCRIPTION
**Pull request by d-w-moore**
Still needs filling out so that the PRC test is running with a complete set of services , ie. iRODS and ICAT nodes.

See https://github.com/irods/irods_testing_jenkins/pull/117 for how this will be run from Jenkins CI

### CODE REVIEW

File: scripts/irods_ping.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
+from getpass import getpass
```
Comment (trel) [inline scripts/irods_ping.py:1]: could just use these?

https://docs.irods.org/4.2.8/system_overview/tips_and_tricks/#monitoring-status-of-irods-servers

or 

https://github.com/irods/irods/blob/a4c97f8a65bd8d2b5d7a505612f2d9d670d33957/scripts/irods/controller.py#L103-L113

Comment (d-w-moore) [reply scripts/irods_ping.py:1]: for the ping, yes... I can use the heartbeat. 

There is still the matter of initializing the credentials. The choice there is
   1. install icommands on the client and use `iinit`
   2. use `irods.password_obfuscation` which comes for free with PRC

Comment (trel) [reply scripts/irods_ping.py:1]: gotcha - yes, PRC is fine fine.   no iinit unless we need it.

File: docker_build/Dockerfile.provider [hunk 2]
```diff
@@ -1,18 +1,26 @@
...
-RUN dpkg -i 
-CMD ./wait_on_condition -i 5 -n 12 "psql -W -h localhost -U postgres -c '\\l' >/dev/null" && \
-    psql -U postgres -f ICAT.sql && \
-    
+
+CMD ./wait_on_condition -i 5 -n 12 su - postgres -c "psql -W -h icat -U postgres -c '\\l' >/dev/null" && \
+    psql -h icat -U postgres -f ICAT.sql && \
+    sed < /var/lib/irods/localhost_setup_postgres.input 's/localhost/icat' \
+        | python /var/lib/irods/scripts/setup_irods.py && \
+    { pgrep -u irods irodsServer >/dev/null || su irods -c '~/irodsctl start'; }  # server does not not auto-start in 4.2.9+
```
Comment (trel) [inline docker_build/Dockerfile.provider:26]: not not

**Comment by d-w-moore** [comment]
Closing in favor of #262

**Pull request closed without merging**